### DISCUSSION TITLE
SRT Push

### DISCUSSION THREAD

**Discussion by stevespaw** [OvenMediaLabs/OvenMediaEngine#1228] [Ideas]
We are really liking this software. We are excited about the ways it will improve our work flow.
We are using SRT input this working perfect so far. We have the API enabled now and are looking to try the push streams next week.
Just one thing would make this solution a game changer.

If we could get lowlatency SRT Caller/Listener push (both Streamid and port method) this would save us a huge amount of complexity in our setup.

You have the UDP MPEG TS so I would think this is an easy lift?

Thanks for this amazing piece of software!

**Comment by getroot** [comment]
Since OvenMediaEnigne depends on ffmpeg for some sub-functions like push, this can be added easily. The problem is that we've been too busy lately. It would be great if someone could PR like MPEGTS push.

**Comment by stevespaw** [comment]
This would be amazing for our workflow, but we don't have c developers to help...

**Comment by stevespaw** [comment]
I envision passthrough video qualities with an API POST with variables:
input:
{vhost/app/stream}
SRT out:
Host: {srthost.com / 1.2.3.4}
port - {9999}
mode - {Caller/Listener/Rendezvous}
if needed - {StreamID}
latency - {200}    convert this from milliseconds, as this is how people see ping times

http://ffmpeg.org/ffmpeg-protocols.html#srt

**Comment by stevespaw** [comment]
I would still like this feature :-)   Any thoughts?

**Reply by getroot** [reply]
As we discussed at NAB I will implement this. ;-)
But I need more time as I still have many more urgent tasks.

**Comment by stevespaw** [comment]
Sounds good ! I was great to meet all of you :-)

**Reply by choigilhoon** [reply]
The SRT Push Publishing feature has been added to the master branch.
Please check this discussion for more details: https://github.com/AirenSoft/OvenMediaEngine/discussions/1299
